generator/src/main/java/com/scottlogic/deg/generator/generation/databags/RowSpecDataBagSourceFactory.java

```diff
@@ -94,6 +95,14 @@ private DataBagSource createReductiveDataBagSourceFor(ReductiveRowSpec rowSpec)
 
         @Override
         public Stream<DataBag> generate(GenerationConfig generationConfig) {
+            //The Reductive walker will emit a single RowSpec that can represent multiple rows
+            //each of these rows will have all fields (except the last one) fixed to a value
+            //if this RowSpec is emitted fully it will give the impression of a set of non-random rows therefore:
+            //emit only one row from the RowSpec then let the walker restart the generation for another random RowSpec
+            if (generationConfig.getDataGenerationType() == GenerationConfig.DataGenerationType.RANDOM){
```

Comment:
Perhaps we need to think about generation type specific classes rather than just using ifs/switches for this sort of check. It's fine for now though

Reply:
Switches are not great, agreed. Not sure how we could refactor this into a strategy or something else at present. Maybe we need to cover this point - and other similar issues - in a larger refactoring ticket

Reply:
This change is still required, even with the recent refactor, as things stand

Reply:
Change refactored into a similar decorated pattern approach